android-core/src/main/java/com/mparticle/networking/MParticleBaseClientImpl.java

```diff
@@ -133,44 +147,86 @@ protected MPUrl getUrl(Endpoint endpoint, @Nullable String identityPath, boolean
-                        .path(subdirectory + mApiKey + "/alias")
+                        .path(subdirectory + pathPostfix)
                         .build();
-                return MPUrl.getUrl(uri.toString(), defaultUrl);
+                return MPUrl.getUrl(uri.toString(), !isDefaultUrl ? generateDefaultURL(uri, defaultDomain, (pathPrefix + pathPostfix)) : null);
             case IDENTITY:
+                pathPrefix = SERVICE_VERSION_1 + "/";
                 subdirectory = overridesSubdirectory ? "" : SERVICE_VERSION_1 + "/";
+                pathPostfix = identityPath;
                 uri = new Uri.Builder()
                         .scheme(BuildConfig.SCHEME)
                         .encodedAuthority(url)
-                        .path(subdirectory + identityPath)
+                        .path(subdirectory + pathPostfix)
                         .build();
-                return MPUrl.getUrl(uri.toString(), defaultUrl);
+                return MPUrl.getUrl(uri.toString(), !isDefaultUrl ? generateDefaultURL(uri, defaultDomain, (pathPrefix + pathPostfix)) : null);
             case AUDIENCE:
+                pathPostfix = SERVICE_VERSION_2 + "/" + mApiKey + "/audience?mpID=" + mConfigManager.getMpid();
                 uri = new Uri.Builder()
                         .scheme(BuildConfig.SCHEME)
                         .encodedAuthority(url)
-                        .path(SERVICE_VERSION_2 + "/" + mApiKey + "/audience?mpID=" + mConfigManager.getMpid())
+                        .path(pathPostfix)
                         .build();
-                return MPUrl.getUrl(uri.toString(), defaultUrl);
+                return MPUrl.getUrl(uri.toString(), !isDefaultUrl ? generateDefaultURL(uri, defaultDomain, pathPostfix) : null);
             default:
                 return null;
         }
     }
 
+    /**
+     * Generates a new URL using the default domain when domain mapping is specified.
+     * This method creates a new URI based on the existing URI, but replaces the domain with the default domain.
+     *
+     * @param uri           The existing URI which includes the domain mapping.
+     * @param defaultDomain The default domain name to be used in the new URI.
+     * @param path          path to be used in the new URI.
+     * @return A new URI with the same path and scheme as the original URI, but with the default domain.
+     */
+    protected MPUrl generateDefaultURL(Uri uri, String defaultDomain, String path) throws MalformedURLException {
+        if (uri != null) {
+            Uri.Builder uriBuilder = Uri.parse(uri.toString()).buildUpon();
+            if (defaultDomain != null && !defaultDomain.isEmpty()) {
+                uriBuilder.encodedAuthority(defaultDomain);
+            }
+            if (path != null && !path.isEmpty()) {
+                uriBuilder.path(path);
+            }
+            return MPUrl.getUrl(uriBuilder.build().toString(), null);
+        }
+        return null;
+    }
+
+    String getPodUrl(
```

Comment:
Is there a reason for this method definition to be split on 5 lines? It seems like the definition on line 204 is longer but is only one line? I know we have a linter that runs, but this seems inconsistent and in general, at least personally, I find this type of formatting harder to read than more compact formatting but that may just be my preference...